### DISCUSSION TITLE
No overwrite signals in multiple mode.

### DISCUSSION THREAD

**Discussion by andbruu** [voxeltycoon/issues#531] [General]
As default, or as an option, I would like if using the multiple mode to place signals never transformed signals already placed. In essence, it should only consider the current block, stopping before any signals leading into, or out of that block.
Currently, I often have to sort out the entrance and exit of the block, as the multiple mode have transformed and/or moved a pre-signal for the upcoming junction.

Only problem I can see, is if I want to change the signals on a long track later (change spacing, direction or type), I would have to remove each signal manually, bore being able to apply my change.
One solution could be a "bulldoze-signals only" tool, or have the multiple mode working as a drag tool.

Perhaps multiple-mode could work as a dynamic block/drag tool. If pressed once it will fill up the current block with the selected signal, stopping before overwriting any previously placed signals. If dragged, it would upon release fill only the section dragged over, but would be allowed to overwrite already placed signals.

**Comment by victorleblais** [comment]
It could use the Shift or Ctrl modifiers, similar to how it works when destroying tracks.

Shift + Buildoze on a signal = delete all signals on this segment of tracks (delimited by intersections)

**Comment by Arsaell** [comment]
The default behaviour of placing signals up to the next intersection makes sense, and it would be totally adequate if there was a "hold ctrl" option to only consider one bloc.

The issue I'm regularly facing, though, is that multiple signal mode WILL change a chain signal into a regular signal, at either end of the track.
This feels wrong: It should upgrade semaphores to signals, and adjust signal intervals, but should never change the type of a signal.

If a player isn't aware of this, they could have a deadlock appearing miles from where they were working because of a signal replacing a chain signal, and have a fun time later getting all those trains out of the way...
